librz/bin/bin.c

```diff
@@ -1080,6 +1080,36 @@ RZ_API void rz_bin_virtual_file_free(RZ_NULLABLE RzBinVirtualFile *vfile) {
 	free(vfile);
 }
 
+/**
+ * \brief Clones a virtual file. Note, the buffer will never be cloned.
+ * This function will assign the buffer to of \p vfile but always sets buf_owned = false;
+ *
+ * \p vfile The virtual to clone.
+ *
+ * \return A clone of the vfile but with clone->buf_owned = false.
+ */
+RZ_API RZ_OWN RzBinVirtualFile *rz_bin_virtual_file_copy(RZ_BORROW RZ_NONNULL RzBinVirtualFile *vfile) {
```

Comment:
```suggestion
RZ_API RZ_OWN RzBinVirtualFile *rz_bin_virtual_file_clone(RZ_BORROW RZ_NONNULL RzBinVirtualFile *vfile) {
```
copy suggests something like memcpy where you provide the destination rather than getting a new instance returned.

Reply:
I would also like to know the use-case for this. Normally you don't need to clone a virtual file since it can be used in multiple maps as-is.

Reply:
I used it in the `mdt` loader. The `mdt` loader sets up it's virtual file instances (for each firmware file once). And then clones it when they are requested by the core. The core expects to own the virtual files retuned, but the plugin also needs an instance of them. So I clone them with the same file buffer, but the cloned version doesn't have the ownership of the buffer. But I would also revert this here. Actually, I forgot why I added them in this PR anyways.

Reply:
Ok, I think it shouldn't be necessary to clone the vfile if the plugin still needs to point to it, but it will be better to late take a closer look again in the pr with the plugin.

Reply:
The main reason for me to keep an instance of the vfile in the plugin, is for future extension.
I'd like to add commands with the bin plugin to do common tasks with the firmware files.

E.g. they have compressed segments. But the destination address for the uncompressed data is not in the ELF or somewhere else. It has to be reverse engineered. I would like to add a command which just takes the offset and then unpacks the segment to a destination.

Another idea would be to patch a segment, compute the hash and patch the hash in the authentication segment (`.b01`). Stuff like this.

But all this requires to keep the instances of the vfiles, sections, segments etc. in the plugin. So they don't need to be parsed again and again.